### DISCUSSION TITLE
How tom explain thread counts

### DISCUSSION THREAD

**Discussion by cforce** [open-telemetry/opentelemetry-collector#9534] [Q&A]
I have executed the otel collector  with hostmetrics  receiver  and all metrics on.
The identification of 10 threads associated with otel process ID raises uncertainty about their origin.

![image](https://github.com/open-telemetry/opentelemetry-collector/assets/393286/2ec64bfa-b13b-46a2-bf1f-c14a89adae4c)

Each thread exhibits approximately 1.7% CPU usage. While the test case assumes a worst-case scenario of 45 req/s through simulating 15 req/s from 3 threads, the chart suggests potential spikes, possibly exceeding 5% CPU usage. 
This test, however, overlooks certain factors such as network infrastructure impact (e.g., network interrupts), inter-process communication overheads, and the load on calling processes using the OpenTelemetry (OTEL) library. Considering these factors, an additional 40-50% may need to be factored in for basic features without host metrics.

Is there  any explanation and a way to easily get from the collector itself which otel thread belongs to which otel module (receiver, processor, exporter, extension):

**Comment by atoulme** [comment]
You can use the pprof extension to get more information, the process.threads metric is what the OS sees.

**Reply by cforce** [reply]
Nice. Will give it a try . The repo proof extension would love to get some screenshots of that web fronted to be expected shown at 127.0.0:11777 😀
